gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopyableFile.java

```diff
@@ -316,23 +316,33 @@ private List<OwnerAndPermission> replicateAncestorsOwnerAndPermission(FileSystem
    */
   public static OwnerAndPermission resolveReplicatedOwnerAndPermission(FileSystem fs, Path path,
       CopyConfiguration copyConfiguration) throws IOException {
-
-    PreserveAttributes preserve = copyConfiguration.getPreserve();
     Optional<FileStatus> originFileStatus = copyConfiguration.getCopyContext().getFileStatus(fs, path);
 
     if (!originFileStatus.isPresent()) {
       throw new IOException(String.format("Origin path %s does not exist.", path));
     }
 
+    return resolveReplicatedOwnerAndPermission(originFileStatus.get(), copyConfiguration);
+  }
+
+  /**
+   * Computes the correct {@link OwnerAndPermission} obtained from replicating source owner and permissions and applying
+   * the {@link PreserveAttributes} rules in copyConfiguration.
+   * @throws IOException
+   */
+  public static OwnerAndPermission resolveReplicatedOwnerAndPermission(FileStatus originFileStatus,
+      CopyConfiguration copyConfiguration) {
+
+    PreserveAttributes preserve = copyConfiguration.getPreserve();
```

Comment:
I want to use the same existing function to calculate the final permission which is  CopyableFile.resolveReplicatedOwnerAndPermission() so that we won't ends up with different behavior in the future when we modify the logic of permission preserving; That function require the whole copyConfiguration because other than the preserve config, we also support target group config to set the user defined group, and I assume that we might want to support something like this for facl as well in the future.